collect_app/src/main/java/org/odk/collect/android/adapters/VersionHidingCursorAdapter.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2012 University of Washington
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.odk.collect.android.adapters;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.view.View;
+import android.widget.SimpleCursorAdapter;
+import android.widget.TextView;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.provider.FormsProviderAPI;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+/**
+ * Implementation of cursor adapter that displays the version of a form if a form has a version.
+ *
+ * @author mitchellsundt@gmail.com
+ */
+public class VersionHidingCursorAdapter extends SimpleCursorAdapter {
+
+    private final Context ctxt;
+    private final ViewBinder originalBinder;
+
+    public VersionHidingCursorAdapter(String versionColumnName, Context context, int layout,
+            Cursor c, String[] from, int[] to) {
+        super(context, layout, c, from, to);
+        ctxt = context;
+        originalBinder = getViewBinder();
+        setViewBinder((view, cursor, columnIndex) -> {
+            String columnName = cursor.getColumnName(columnIndex);
+            if (columnName.equals(FormsProviderAPI.FormsColumns.DATE) || columnName.equals(FormsProviderAPI.FormsColumns.MAX_DATE)) {
```

Comment:
Adding `MAX_DATE` to the API explicitly seems odd to me because it's not a column name. Can you leave the call on `max` in `FormsProvider` like you had it before and do a check for `columnName.contains(FormsColumns.MAX_DATE)` here? It's still odd because a call on `.getColumnName` returns a function call but at least it's isolated here.

Reply:
I added a comment that it is used only to get latest forms for each form_id.
`columnName` is a String so I can't call `contains()` on it I could use `cursor.getColumnNames()` but it returns String[] so also suing  `contains()` is not possible I would need to convert it and it would require more code.
If could just remove the last commit at all and use `MAX(" + FormsProviderAPI.FormsColumns.DATE + ")` but I don't think it's better.

Reply:
> columnName is a String so I can't call contains() on it

https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#contains-java.lang.CharSequence- Am I misunderstanding?

Reply:
Ok got it I thought you mean contains() on the array of columns to determine if it contains `date` or `MAX(date)`... 
but still, I will need to use `"MAX(" + FormsColumns.DATE + ")"` not only in FormsProvider but also here https://github.com/opendatakit/collect/pull/3193/files#diff-af25ce4b576502ac52785c5d4cd8a844R174

isn't it more confusing?

Reply:
I don't feel strongly about it.